src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1042,7 +1042,7 @@ abstract class RefChecks extends Transform {
       // equals.
       def isUsingWarnableEquals = {
         val m = receiver.info.member(nme.equals_)
-        ((m == Object_equals) || (m == Any_equals) || isMethodCaseEquals(m))
+        (m == Object_equals) || (m == Any_equals) || isMethodCaseEquals(m)
```

Comment:
the diff shows that you deleted the inner parens. You probably intended to delete the outer parens. (I am trying out some Swiss humour.)

Reply:
shoot!